### PR TITLE
hetzner-kube: 0.3.1 -> 0.3.2rc1

### PR DESCRIPTION
**Pull request by offlinehacker**
hetzner-kube: 0.3.1 -> 0.3.2rc1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
Why a release candidate?

**Comment by offlinehacker** [comment]
Well their versioning is a bit weird: https://github.com/xetys/hetzner-kube/releases

File: pkgs/applications/networking/cluster/hetzner-kube/default.nix [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-
-buildGoPackage {
+buildGoPackage rec {
   name = "hetzner-kube-${version}";
+  version = "0.3.2-rc1";
 
   src = fetchFromGitHub {
     owner = "xetys";
     repo = "hetzner-kube";
     rev = "${version}";
```
Comment (worldofpeace) [inline pkgs/applications/networking/cluster/hetzner-kube/default.nix]:
```suggestion
    rev = version;
```

File: pkgs/applications/networking/cluster/hetzner-kube/default.nix [hunk 2]
```diff
@@ -1,18 +1,19 @@
 { lib, buildGoPackage, fetchFromGitHub, ... }:
 
-let version = "0.3.1"; in
-
-buildGoPackage {
+buildGoPackage rec {
   name = "hetzner-kube-${version}";
```
Comment (worldofpeace) [inline pkgs/applications/networking/cluster/hetzner-kube/default.nix]:
```suggestion
  pname = "hetzner-kube";
```

**Comment by worldofpeace** [comment]
> Well their versioning is a bit weird: https://github.com/xetys/hetzner-kube/releases

Indeed that is kinda weird. Especially their use of `RC`, and certain releases being `pre-releases`.
I can see you went with the one labeled `Latest Release` but I don't get why `4.x.x.*` wouldn't be a stable release.

Perhaps open on issue?

**Comment by offlinehacker** [comment]
@GrahamcOfBorg build

**Comment by offlinehacker** [comment]
@GrahamcOfBorg build hetzner-kube

**Comment by offlinehacker** [comment]
@worldofpeace yeah agreed, i opened issue: https://github.com/xetys/hetzner-kube/issues/252

Currently i'm not using `hetzner-kube`, so i can't really say what is working in what version.

File: pkgs/applications/networking/cluster/hetzner-kube/default.nix [hunk 3]
```diff
@@ -1,18 +1,20 @@
 { lib, buildGoPackage, fetchFromGitHub, ... }:
 
-let version = "0.3.1"; in
-
-buildGoPackage {
-  name = "hetzner-kube-${version}";
+buildGoPackage rec {
+  name = "${pname}-${version}";
```
Comment (worldofpeace) [inline pkgs/applications/networking/cluster/hetzner-kube/default.nix]: This is done default now in `stdenv.mkDerivation` and extends to `buildGoPackage`

**Pull request merged**